app/src/main/java/emu/skyline/preference/IntegerListPreference.kt

```diff
@@ -0,0 +1,329 @@
+            attrs, sR.styleable.IntegerListPreference, defStyleAttr, defStyleRes
+        )
+
+        entries = TypedArrayUtils.getTextArray(
+            a, sR.styleable.IntegerListPreference_entries,
+            sR.styleable.IntegerListPreference_android_entries
+        )
+
+        val entryValuesId = TypedArrayUtils.getResourceId(
+            a, sR.styleable.IntegerListPreference_android_entryValues,
+            sR.styleable.IntegerListPreference_android_entryValues, 0
+        )
+        entryValues = if (entryValuesId != 0) res.getIntArray(entryValuesId) else null
+
+        if (TypedArrayUtils.getBoolean(
+                a, sR.styleable.IntegerListPreference_useSimpleSummaryProvider,
+                sR.styleable.IntegerListPreference_useSimpleSummaryProvider, false
+            )
+        ) {
+            summaryProvider = SimpleSummaryProvider.instance
+        }
+
+        refreshRequired = TypedArrayUtils.getBoolean(
+            a, sR.styleable.IntegerListPreference_refreshRequired,
+            sR.styleable.IntegerListPreference_refreshRequired, false
+        )
+
+        a.recycle()
+    }
+
+    /**
+     * @param entriesResId The entries array as a resource
+     */
+    fun setEntries(@ArrayRes entriesResId : Int) {
+        entries = context.resources.getTextArray(entriesResId)
+    }
+
+    /**
+     * @param entryValuesResId The entry values array as a resource
+     */
+    fun setEntryValues(@ArrayRes entryValuesResId : Int) {
+        entryValues = context.resources.getIntArray(entryValuesResId)
+    }
+
+    /**
+     * @return The entry corresponding to the current value, or `null`
+     */
+    fun getEntry() : CharSequence? {
+        return entries?.getOrNull(getValueIndex())
+    }
+
+    /**
+     * @param value The value whose index should be returned
+     * @return The index of the value, or -1 if not found
+     */
+    private fun findIndexOfValue(value : Int?) : Int {
+        entryValues?.let {
+            if (value != null) {
+                for (i in it.indices.reversed()) {
+                    if (it[i] == value) {
+                        return i
+                    }
+                }
+            }
+        }
+        return value ?: -1
+    }
+
+    /**
+     * Sets the value to the given index from the entry values.
+     * @param index The index of the value to set
+     */
+    fun setValueIndex(index : Int) {
+        value = entryValues?.get(index) ?: index
+    }
+
+    private fun getValueIndex() : Int {
+        return findIndexOfValue(value)
+    }
+
+    override fun onGetDefaultValue(a : TypedArray, index : Int) : Any? {
+        return a.getInt(index, 0)
+    }
+
+    override fun onSetInitialValue(defaultValue : Any?) {
+        // `Preference` superclass passes a null defaultValue if it is sure there
+        // is already a persisted value, se we have to account for that here by
+        // passing a random number as default value.
```

Comment:
Don't break any comments into newlines mid-sentence and don't make it this short either, most displays can show far more text per-line than this esp. when combined with wrapping, as a result of this you should treat line breaks like paragraphs and not just arbitrary spacing. 

There's more cases of this in this file, do fix those as well.